featuretools/tests/primitive_tests/test_feature_serialization.py

```diff
@@ -94,15 +95,23 @@ def test_pickle_features_with_custom_primitive(pd_es, tmpdir):
 
 def test_serialized_renamed_features(es):
     def serialize_name_unchanged(original):
-        renamed = original.rename('MyFeature')
-        assert renamed.get_name() == 'MyFeature'
+        def check_names(feature, name):
+            assert feature.get_name() == name
+            if len(names) == 1:
```

Comment:
this section should test that `feature.get_feature_names()` matches the expected `names` value

instead of generating the expected `names` value inside of `check_names`, we could probably do it outside like we do with `new_name`, and pass it in

Reply:
Is there a reason we would want to though? We have a clear naming schema where `_names` depends on `_name`. Passing `names` in would produce repeated code outside the function

Reply:
That explanation was missing some points, let me rephrase

* Currently, `names` variable in the function does not depend on `feature`.  It is always equal to `renamed.get_feature_names()`, even if `check_names` is testing `deserialized`
* If we moved the if/else for determining the expected `names` value outside of the function and passed the the result in it wouldn't create repeated code but it would add an extra line.  Let's leave it as is.